enumerables.rb

```diff
@@ -1,41 +1,106 @@
+# frozen_string_literal: true
+# My each method for Ruby, you can use it by simply calling it .my_each
 module Enumerable
-    def my_each
-      return to_enum(:my_each) unless block_given?
-      
-      i = 0
-      array = to_a
-      while i < array.length
-        yield(array[i])
-        i+=1
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    array = to_a
+    while i < array.length
+      yield(array[i])
+      i += 1
     end
     array
   end
- 
 
-def my_each_with_index
+  def my_each_with_index
     return to_enum(:my_each_with_index) unless block_given?
-    
+
     i = 0
     array = to_a
     while i < array.length
       yield(array[i], i)
-      i+=1
+      i += 1
+    end
+    array
   end
-  array
-end
 
-def my_select
+  def my_select
     return to_enum(:my_select) unless block_given?
+
     new_array = []
-  my_each do |x|
-new_array.push(x) if yield(x) == true
+    my_each do |x|
+      new_array.push(x) if yield(x) == true
+    end
+    new_array
   end
-  new_array
-end
-end
-  
 
-puts [1,2,3,4,5,6].my_select(&:even?)
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
 
+    my_each do |x|
+      return false if yield(x) != true
+    end
+    true
+  end
```

Comment:
### #my_all?
- [ ] When no block or argument is given returns true when none of the collection members are false or nil
- [ ] When no block or argument is given returns false when one of the collection members is false or nil
- [ ] When a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] When a class is passed as an argument returns false if any of the collection is not a member of such class
- [ ] When a Regex is passed as an argument returns false if any of the collection does not match the Regex
- [ ] When a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern